base_layer/wallet/src/transaction_service/protocols/transaction_receive_protocol.rs

```diff
@@ -430,6 +339,29 @@ where TBackend: TransactionBackend + Clone + 'static
                             self.id,
                             TransactionServiceError::TransactionCancelled,
                         ));
+                    },
+                    () = resend_timeout => {
+                        if let Err(e) = send_transaction_reply(
+                            inbound_tx.clone(),
+                            self.resources.outbound_message_service.clone(),
+                            self.resources.config.direct_send_timeout,
+                        )
+                        .await
+                        {
+                            warn!(
+                                target: LOG_TARGET,
+                                "Error resending Transaction Reply (TxId: {}): {:?}", self.id, e
+                            );
+                        } else {
+                            self.resources
+                                .db
+                                .increment_send_count(self.id)
+                                .await
+                                .map_err(|e| TransactionServiceProtocolError::new(self.id, TransactionServiceError::from(e)))?;
+                        }
```

Comment:
`match` is more idiomatic

```suggestion
                        match send_transaction_reply(
                            inbound_tx.clone(),
                            self.resources.outbound_message_service.clone(),
                            self.resources.config.direct_send_timeout,
                        )
                        .await
                        {
                        Ok(_) => {
                           self.resources
                                .db
                                .increment_send_count(self.id)
                                .await
                                .map_err(|e| TransactionServiceProtocolError::new(self.id, TransactionServiceError::from(e)))?;
                        },
                        Err(err) => {
                            warn!(
                                target: LOG_TARGET,
                                "Error resending Transaction Reply (TxId: {}): {:?}", self.id, e
                            );
                        
                        }
```

Reply:
Damnit Stanley, why do you hate the classic C syntax so much?

Reply:
But ok will update but honestly its because the classic syntax is easier to format manually inside a select! macro

Reply:
Yeess .. submit to the will of rust